pyxem/signals/polar_diffraction2d.py

```diff
@@ -256,6 +261,43 @@ def get_resolved_pearson_correlation(
 
         return correlation
 
+    def subtract_diffraction_background(
+        self, method="radial median", inplace=False, **kwargs
+    ):
+        """Background subtraction of the diffraction data.
+
+        Parameters
+        ----------
+        method : str, optional
+            'radial median', 'radial percentile'
+            Default 'radial median'.
+
+            For 'radial median' no extra parameters are necessary.
+
+            For 'radial percentile' the 'percentile' argument decides 
+            which percentile to substract.
+        **kwargs :
+                To be passed to the chosen method.
+
+        Returns
+        -------
+        s : PolarDiffraction2D or LazyPolarDiffraction2D signal
+
+        """
+        method_dict = {
+            "radial median": _polar_subtract_radial_median,
+            "radial percentile": _polar_subtract_radial_percentile,
+        }
+        if method not in method_dict:
+            raise NotImplementedError(
+                f"The method specified, '{method}',"
+                f" is not implemented.  The different methods are:  "
+                f"{', '.join(method_dict.keys())}."
+            )
+        subtraction_function = method_dict[method]
+
+        return self.map(subtraction_function, inplace=inplace, **kwargs)
```

Comment:
It might be nice to set the `output_dtype = self.data.dtype` and the `output_signal_size=self.axes_manager._signal_shape_in_array`

That just stops it from "testing" what the output shape and datatype is which makes things a bit faster.